thirtyinch/src/test/java/net/grandcentrix/thirtyinch/internal/PresenterScopeTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2017 grandcentrix GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.grandcentrix.thirtyinch.internal;
+
+import net.grandcentrix.thirtyinch.TiPresenter;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.fail;
+
+public class PresenterScopeTest {
+
+    @Test
+    public void addOneMapping() throws Exception {
+
+        final PresenterScope scope = new PresenterScope();
+        final TiPresenter presenter = new TiPresenter() {
+        };
+        scope.save("a", presenter);
+
+        assertThat(scope.getAll())
+                .hasSize(1)
+                .contains(presenter);
+
+        assertThat(scope.size()).isEqualTo(1);
+        assertThat(scope.getAllMappings()).hasSize(1);
+        assertThat(scope.getAllMappings().get(0).getKey()).isEqualTo("a");
+        assertThat(scope.getAllMappings().get(0).getValue()).isEqualTo(presenter);
+    }
+
+    @Test
+    public void addTwoMappings() throws Exception {
+
+        final PresenterScope scope = new PresenterScope();
+        final TiPresenter presenter1 = new TiPresenter() {
+        };
+        scope.save("a", presenter1);
+
+        // override with same id
+        final TiPresenter presenter2 = new TiPresenter() {
```

Comment:
We don't need to create another presenter here.. 
Just reuse presenter1

Reply:
Oh that's a good edge case. We should prevent adding a presenter twice with a different id.